src/main/java/net/clgd/ccemux/emulation/EmulatedPalette.java

```diff
@@ -4,41 +4,74 @@
 import java.util.List;
 
 import dan200.computercraft.shared.util.Palette;
+import lombok.Getter;
 
 public class EmulatedPalette extends Palette {
 	@FunctionalInterface
-	public interface Listener {
+	public interface ColorChangeListener {
 		void setColour(int index, double r, double g, double b);
 	}
 
 	private final Palette delegate;
 
-	private final List<Listener> listeners = new ArrayList<>();
+	private final List<ColorChangeListener> listeners = new ArrayList<>();
+
+	@Getter
+	private boolean changed = true;
 
 	public EmulatedPalette(Palette delegate) {
 		this.delegate = delegate;
 	}
 
+	public void setChanged() {
+		changed = true;
+	}
+
+	public void clearChanged() {
+		changed = false;
```

Comment:
I think a `setChanged(boolean)` would be more intuitive.

Reply:
This is just to follow the convention used by `Terminal` but I can change it